File: app/src/main/java/com/scheffer/erik/popularmovies/MovieDatabaseApi/Movie.java
package com.scheffer.erik.popularmovies.MovieDatabaseApi;

import android.os.Parcel;
import android.os.Parcelable;

public class Movie implements Parcelable {
    private String title;
    private float voteAverage;
    private String overview;
    private String posterPath;
    private String releaseDate;

    private Movie(Parcel in) {
        title = in.readString();
        voteAverage = in.readFloat();
        overview = in.readString();
        posterPath = in.readString();
        releaseDate = in.readString();
    }

    public static final Creator<Movie> CREATOR = new Creator<Movie>() {
        @Override
        public Movie createFromParcel(Parcel in) {
            return new Movie(in);
        }

        @Override
        public Movie[] newArray(int size) {
            return new Movie[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(title);
        dest.writeFloat(voteAverage);
        dest.writeString(overview);
        dest.writeString(posterPath);
        dest.writeString(releaseDate);
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public float getVoteAverage() {
        return voteAverage;
    }

    public void setVoteAverage(float voteAverage) {
        this.voteAverage = voteAverage;
    }

    public String getOverview() {
        return overview;
    }

    public void setOverview(String overview) {
        this.overview = overview;
    }

    public String getPosterPath() {
        return posterPath;
    }

    public void setPosterPath(String posterPath) {
        this.posterPath = posterPath;
    }

    public String getReleaseDate() {
        return releaseDate;
    }

    public void setReleaseDate(String releaseDate) {
        this.releaseDate = releaseDate;
    }

    @Override
    public String toString() {
        return "Movie{" +
                "title='" + title + '\'' +
                ", voteAverage=" + voteAverage +
                ", overview='" + overview + '\'' +
                ", posterPath='" + posterPath + '\'' +
                '}';
    }
}

File: app/src/main/java/com/scheffer/erik/popularmovies/moviedatabaseapi/models/Movie.kt
package com.scheffer.erik.popularmovies.moviedatabaseapi.models

import android.os.Parcelable
import com.raizlabs.android.dbflow.annotation.PrimaryKey
import com.raizlabs.android.dbflow.annotation.Table
import com.scheffer.erik.popularmovies.database.AppDatabase
import kotlinx.android.parcel.Parcelize
import java.util.*

@Parcelize
@Table(database = AppDatabase::class, allFields = true)
data class Movie(@PrimaryKey(autoincrement = true) var databaseId: Long = 0,
                 var id: Long = 0,
                 var title: String = "",
                 var voteAverage: Float = 0F,
                 var overview: String = "",
                 var posterPath: String = "",
                 var releaseDate: Date = Date()) : Parcelable

data class MovieResultList(val results: List<Movie>)